### PR TITLE
wip implementation work based on 1.x branch

### PR DESCRIPTION
**Pull request by boneskull**
- add coverage for most implemented non-`npm install` behavior
- add a wallabyjs config file
- add [`read-package-json`](https://npm.im/read-package-json) for some normalization
- ~~rename `test/index.js` to `test/index.spec.js` because it changes the file icon in vscode~~
- rename fixtures, reorganize tests somewhat
- add a helper function to run `createPackageJson` in silent mode w/ `barePrompt`
- change some assertions to use `deepStrictEqual`
- add TODO list to `README.md`
- better handling of `scripts`, `keywords`, and `repository` fields
- update build matrix

### CODE REVIEW

**Comment by boneskull** [comment]
should probably revert the rename of `test/index.js` cause the diff is bad.

**Comment by boneskull** [comment]
@wesleytodd `Object.fromEntries` is not available in Node.js v10.x.  So if we want to support that version of Node (probably should...) we'd need to pull in a ponyfill or just punt.  I punted.

File: index.js [hunk 1]
```diff
@@ -258,21 +261,27 @@ async function format (opts, pkg = {}) {
   pkg.type = opts.type || 'commonjs'
 
   if (opts.keywords && opts.keywords.length) {
-    pkg.keywords = opts.keywords
+    // TODO: extra parsing going on here due to wesleytodd/opta#1
+    pkg.keywords = uniquify([...(pkg.keywords || []), ...parseList(opts.keywords)])
```
Comment (wesleytodd) [inline index.js:265]: great name :P

File: index.js [hunk 2]
```diff
@@ -240,9 +242,10 @@ async function readPackageJson (options, { log } = {}) {
     type: pkg.type,
     author: pkg.author,
     description: pkg.description,
-    repository: pkg.repository && pkg.repository.url,
```
Comment (wesleytodd) [inline index.js:243]: This is a good change, I remember writing this and not liking it.

**Review by wesleytodd — APPROVED** [review]
Looks good.  Do you plan to just keep adding more features onto this or as separate PRs?

On the `Object.entries` side, yeah this has caught me a few times.  Good changes on that.

**Comment by wesleytodd** [comment]
Oh, also what does `wallabyjs` do?  I have not used it before.

**Comment by boneskull** [comment]
wallabyjs.com - instant-feedback test runner integrated with IDEs. it’s very handy; better than any framework’s “watch” mode. config file is useful for those who use it, but i can rip it out if you’d rather not have the clutter

**Comment by wesleytodd** [comment]
> config file is useful for those who use it, but i can rip it out if you’d rather not have the clutter

Doesn't matter much to me if it is useful to people.

**Comment by wesleytodd** [comment]
Hey, I know this is marked as draft, but do you see any reason not to start merging this as the work is done?  I am going to integrate this into a project today, so figured it would be a good time for me to merge at least base on top of your work.

**Comment by wesleytodd** [comment]
I added you as a collaborator here (I should have done that in the first place, but better late then never) and pushed your branch.  We can discuss what to do with the PR since I cannot find a way to just swap out the repo/branch in this way, but I will probably have a few additions today as I mess with it.

**Pull request merged**